src/linker/Linker.Dataflow/XmlFlowAnnotationSource.cs

```diff
@@ -0,0 +1,690 @@
+				}
+			}
+			if (returnAnnotation != 0 || parameterAnnotation.Count > 0)
+				_methods[method] = new AnnotatedMethod (returnAnnotation, parameterAnnotation.ToArray ());
+
+		}
+
+		ArrayBuilder<(string, ArrayBuilder<Attribute>)> ProcessParameters (TypeDefinition type,
+			MethodDefinition method, XPathNodeIterator iterator)
+		{
+			var methodParameters = new ArrayBuilder<(string, ArrayBuilder<Attribute>)> ();
+			while (iterator.MoveNext ()) {
+				methodParameters.Add ((GetAttribute (iterator.Current, "name"), ProcessAttributes (iterator.Current.SelectChildren ("attribute", _ns))));
+			}
+			return methodParameters;
+		}
+
+		ArrayBuilder<ArrayBuilder<Attribute>> ProcessReturnParameters (TypeDefinition type,
+			MethodDefinition method, XPathNodeIterator iterator)
+		{
+			var methodParameters = new ArrayBuilder<ArrayBuilder<Attribute>> ();
+
+			while (iterator.MoveNext ()) {
+				methodParameters.Add (ProcessAttributes (iterator.Current.SelectChildren ("attribute", _ns)));
+			}
+			return methodParameters;
+		}
+
+
+		void ProcessMethodIfNotNull (TypeDefinition type, MethodDefinition method, XPathNodeIterator iterator)
+		{
+			if (method == null)
+				return;
+
+			ProcessMethodChildren (type, method, iterator);
+		}
+
+		void ProcessMethodName (TypeDefinition type, string name, XPathNodeIterator iterator)
+		{
+			if (!type.HasMethods)
+				return;
+
+			foreach (MethodDefinition method in type.Methods)
+				if (name == method.Name)
+					ProcessMethodChildren (type, method, iterator);
+		}
+
+		protected static MethodDefinition GetMethod (TypeDefinition type, string signature)
+		{
+			if (type.HasMethods)
+				foreach (MethodDefinition meth in type.Methods)
+					if (signature == GetMethodSignature (meth, false))
+						return meth;
+
+			return null;
+		}
+
+		public static string GetMethodSignature (MethodDefinition meth, bool includeGenericParameters)
+		{
+			StringBuilder sb = new StringBuilder ();
+			sb.Append (meth.ReturnType.FullName);
+			sb.Append (" ");
+			sb.Append (meth.Name);
+			if (includeGenericParameters && meth.HasGenericParameters) {
+				sb.Append ("`");
+				sb.Append (meth.GenericParameters.Count);
+			}
+
+			sb.Append ("(");
+			if (meth.HasParameters) {
+				for (int i = 0; i < meth.Parameters.Count; i++) {
+					if (i > 0)
+						sb.Append (",");
+
+					sb.Append (meth.Parameters[i].ParameterType.FullName);
+				}
+			}
+			sb.Append (")");
+			return sb.ToString ();
+		}
+
+		void ProcessProperties (TypeDefinition type, XPathNodeIterator iterator)
+		{
+			while (iterator.MoveNext ())
+				ProcessProperty (type, iterator);
+		}
+
+		protected virtual void ProcessProperty (TypeDefinition type, XPathNodeIterator iterator)
+		{
+#if !FEATURE_ILLINK
+			if (IsExcluded (iterator.Current))
+				return;
+#endif
+
+			string value = GetSignature (iterator.Current);
+			if (!String.IsNullOrEmpty (value))
+				ProcessPropertySignature (type, value, GetAccessors (iterator.Current), iterator);
```

Comment:
There's no test coverage for this. While yes, technically one could have properties with the same name that only differ by their type, I'm not sure if we really need to have syntax to express this. Mainstream .NET languages don't support this.

Reply:
I did not change this behavior, I think it should be there at least as a way to keep the format being used in ResolveFromXmlStep